### PR TITLE
handle statefulsets watchers for  k8s versions > 16

### PR DESCRIPTION
**Pull request by mohamedveron**

### CODE REVIEW

File: proc/watcher.go [hunk 1]
```diff
@@ -324,37 +324,70 @@ func (observer *Observer) watchStatefulSets(
 ) {
 
 	infof(nil, "{kubernetes} starting observer of statefulSets")
+	if client.APIVersion().Version == "v1" {
```
Comment (khaledkbadr) [inline proc/watcher.go:327]: this is the client API version, it's not the server's version :)

Comment (mohamedveron) [reply proc/watcher.go:327]: the issue related with the api version not server version

Comment (mohamedveron) [reply proc/watcher.go:327]: panic: interface conversion: interface {} is *v1.StatefulSet, not *v1beta2.StatefulSet [recovered]
panic: interface conversion: interface {} is *v1.StatefulSet, not *v1beta2.StatefulSet [recovered]
panic: interface conversion: interface {} is *v1.StatefulSet, not *v1beta2.StatefulSet

Comment (mohamedveron) [reply proc/watcher.go:327]: from logs

Comment (khaledkbadr) [reply proc/watcher.go:327]: yeah, it can't match data coming from the server with data in the client. this comes from the `reflect` module in k8s client

Comment (khaledkbadr) [reply proc/watcher.go:327]: I doubled check, yeah the issue came from https://github.com/MagalixCorp/magalix-agent/pull/47/files#diff-b3bfb4a5cc3a668c227740705f21e129L333. PR should fix it alright

**Pull request merged**